collection.go

```diff
@@ -650,7 +650,14 @@ func (c *Collection) begin(db DB) (*pg.Tx, int64, error) {
 	_, err = tx.Exec("LOCK TABLE ?", pg.Q(c.tableName))
 	if err != nil {
 		_ = tx.Rollback()
-		return nil, 0, err
+		if strings.Contains(err.Error(), "syntax error at or near \"lock\"") {
+			tx, err = db.Begin()
+			if err != nil {
+				return nil, 0, err
+			}
+		} else {
+			return nil, 0, err
+		}
```

Comment:
Please add a comment that `LOCK TABLE` is not supported in CockroackDB. Also you could swap conditions to use early return:

```
if !if strings.Contains() {
    return
}

db.Begin
```